lib/livebook/runtime/evaluator.ex

```diff
@@ -620,6 +620,73 @@ defmodule Livebook.Runtime.Evaluator do
     end
   end
 
+  defp eval("erlang", code, binding, env) do
+    try do
+      erl_binding =
+        binding
+        |> List.foldl(:erl_eval.new_bindings(), fn {name, value}, erl_binding ->
+          :erl_eval.add_binding(
+            name
+            |> :erlang.atom_to_binary()
+            |> Macro.camelize()
+            |> :erlang.binary_to_atom(),
+            value,
+            erl_binding
+          )
+        end)
+
+      {:ok, tokens, _} = :erl_scan.string(code |> String.to_charlist())
+      {:ok, parsed} = :erl_parse.parse_exprs(tokens)
+
+      {:value, result, new_erl_binding} =
+        :erl_eval.exprs(
+          parsed,
+          erl_binding,
+          {:eval, &erl_local_func/3}
+        )
+
+      binding =
+        new_erl_binding
+        |> Keyword.drop(Keyword.keys(erl_binding))
+        |> List.foldl(binding, fn {name, value}, binding ->
+          binding
+          |> Keyword.put(
+            name
+            |> :erlang.atom_to_binary()
+            |> Macro.underscore()
+            |> :erlang.binary_to_atom(),
+            value
+          )
+        end)
+
+      {:ok, result, binding, env}
+    catch
+      kind, error ->
+        stacktrace = prune_stacktrace(__STACKTRACE__)
+
+        code_error =
+          if code_error?(error) and (error.file == env.file and error.file != "nofile") do
+            %{line: error.line, description: error.description}
+          else
+            nil
+          end
+
+        {:error, kind, error, stacktrace, code_error}
+    end
+  end
+
+  defp erl_local_func(:f, [], _bindings) do
+    {:value, :ok, :erl_eval.new_bindings()}
+  end
+
+  defp erl_local_func(:f, [{:var, _, name}], bindings) do
+    {:value, :ok, :erl_eval.del_binding(name, bindings)}
+  end
+
+  defp erl_local_func(other, _args, _bindings) do
+    :erlang.error(:function_clause, other)
```

Comment:
Function clause or undefined function>?

Reply:
Actually, we might not need this one at all (or with different semantics). It implements the erlang shell's `f/0` and `f/1` pseudo-functions (which would only work if the "merge" step was amended with deletion).

Reply:
Yeah, I am fine with removing this for now. Removing bindings may be something that requires special consideration from the Livebook side (and it may be it doesn't work at all right now).